libandroid-navigation/src/main/java/org/maplibre/navigation/android/navigation/v5/models/BannerComponents.kt

```diff
@@ -0,0 +1,242 @@
+     *
+     * @since 5.0.0
+     */
+    @SerialName("imageURL")
+    val imageUrl: String?,
+
+    /**
+     * A List of directions indicating which way you can go from a lane
+     * (left, right, or straight). If the value is ['left', 'straight'],
+     * the driver can go straight or left from that lane.
+     * Present if this is a lane component.
+     *
+     * @since 3.2.0
+     */
+    val directions: List<String>?,
+
+    /**
+     * A boolean telling you if that lane can be used to complete the upcoming maneuver.
+     * If multiple lanes are active, then they can all be used to complete the upcoming maneuver.
+     * Present if this is a lane component.
+     *
+     * @since 3.2.0
+     */
+    val active: Boolean?,
+) {
+
+    fun toBuilder(): Builder {
+        return Builder(
+            text = text,
+            type = type
+        ).apply {
+            withSubType(subType)
+            withAbbreviation(abbreviation)
+            withAbbreviationPriority(abbreviationPriority)
+            withImageBaseUrl(imageBaseUrl)
+            withImageUrl(imageUrl)
+            withDirections(directions)
+            withActive(active)
+        }
+    }
+
+    enum class Type(val text: String) {
+        /**
+         * Default. Indicates the text is part of the instructions and no other type.
+         *
+         * @since 3.0.0
+         */
+        @SerialName("text")
+        TEXT("text"),
+
+        /**
+         * This is text that can be replaced by an imageBaseURL icon.
+         *
+         * @since 3.0.0
+         */
+        @SerialName("icon")
+        ICON("icon"),
+
+        /**
+         * This is text that can be dropped, and should be dropped if you are rendering icons.
+         *
+         * @since 3.0.0
+         */
+        @SerialName("delimiter")
+        DELIMITER("delimiter"),
+
+        /**
+         * Indicates the exit number for the maneuver.
+         *
+         * @since 3.0.0
+         */
+        @SerialName("exit-number")
+        EXIT_NUMBER("exit-number"),
+
+        /**
+         * Provides the the word for exit in the local language.
+         *
+         * @since 3.0.0
+         */
+        @SerialName("exit")
+        EXIT("exit"),
+
+        /**
+         * Indicates which lanes can be used to complete the maneuver.
+         *
+         * @since 3.0.0
+         */
+        @SerialName("lane")
+        LANE("lane"),
+
+        /**
+         * This view gives guidance through junctions and is used to complete maneuvers.
+         */
+        @SerialName("guidance-view")
+        GUIDANCE_VIEW("guidance-view"),
+
+        /**
+         * This view gives guidance through signboards and is used to complete maneuvers.
+         */
+        @SerialName("signboard")
+        SIGNBOARD("signboard"),
+
+        /**
+         * This view gives guidance through junctions and is used to complete maneuvers.
+         */
+        @SerialName("jct")
+        JCT("jct")
+    }
+
+    class Builder(
```

Comment:
Why do have some models a builder other not? And do we need a builder with named parameters at all?

Reply:
The problem with the data classes is the Java code. The constructor and the copy functions needs all parameters (without naming support).

As our`ui` module will keeps in Java for now, we need a simple way to access and modify the models. I only create builders for our needs in the `ui` module.

A only for Kotlin deprecation is not possibile. I would remove these builders, if the `ui` has also converted. If you think there are a lot of Java clients using this library, we should thinking about to add missing builders to the other models.

Reply:
Ok, I am fine with Builders as well. I am not sure how many Java clients there are left hat actively use this project except for people using the ui module.

So I think I'd vote for not creating Builders, but since we have them now already, I don't think we need to delete them, but maybe phase them out for the next major release? 

For the UI module I would be fine to only convert a few classes to Kotlin to avoid too much ugliness due to the Builders. But yeah, your decision. I don't have a strong opinion either way.

Reply:
While the conversation can also break some code, and the tests are very deep nested, I will keep it as it is. We can have a look on this, in a follow up (#128).